### PR TITLE
feat: add code execution status to chat messages.

### PR DESCRIPTION
**Pull request by EtiennePerot**
This PR adds `code_executions` as an array of code execution statuses to chat messages. The intent of this data is to be displayed in a similar manner as citations: at the bottom of the message, with buttons that open a modal for more info. As a point of reference, here is how similar data looks in ChatGPT when it executes code:

**ChatGPT code execution status**:

Screenshot from 2024-09-10 23-08-47

**ChatGPT code execution modal**:

Screenshot from 2024-09-10 23-09-06

**Why not use the existing "citations" modals?** Code execution data doesn't fit well in that, because it fundamentally differs in its formatting. Code execution status includes the code that was run (which benefits from being syntax-highlighted), and the output and generated files. This differs from citations which are just list of document names and links.

Additionally, code execution is a process, whereas citations are only emitted once. This is why code execution data uses an ID-based approach, where each code execution instance is identified by a unique ID and can be updated by emitting a new `code_execution` message with the same ID. This allows the code execution status to be updated as code runs.

# Pull Request Checklist

**Before submitting, make sure you've checked the following:**

- [x] **Target branch:** `dev`
- [x] **Description:** See above.
- [x] **Changelog:** OK.
- [x] **Documentation:** No documentation to update.
- [x] **Dependencies:** No new dependencies.
- [x] **Testing:** Manually tested to produce buttons similar to citations at the bottom of messages.
- [x] **Code review:** OK.
- [x] **Prefix:** `feat`

# Changelog Entry

 ### Description

Citation messages can now be used to convey code execution status.

### Added

* New type of citation to display code execution status.

### Screenshots or Videos

Code execution video

### CODE REVIEW

File: src/lib/components/chat/Chat.svelte [hunk 1]
```diff
@@ -179,6 +179,21 @@
 				} else {
 					message.citations = [data];
 				}
+			} else if (type === 'code_execution') {
```
Comment (tjbck) [inline src/lib/components/chat/Chat.svelte:182]: Perhaps it's better to use the existing 'citations' event type and add a new 'code_execution' sub data type here.

Comment (EtiennePerot) [reply src/lib/components/chat/Chat.svelte]: Done.

**Comment by EtiennePerot** [comment]
PR updated to include the modal dialog for code execution status. Also added video of the feature.

**Comment by tjbck** [comment]
LGTM, Thanks!

**Comment by tjbck** [comment]
@EtiennePerot did some refac (notably uuid -> id, but mostly styling changes), testing wanted with dev here!

**Comment by EtiennePerot** [comment]
Looks great and works great, thanks! Tested from `dev`.

**Pull request merged**